### ISSUE TITLE
For posts with no usable content, do not trust their content embeddings

### ISSUE THREAD

**Issue by raindrift** [greenearth-social/api#109]
I think we get content embeddings, even for posts that have no truly usable content.

The embedding should be over the post content, alt text for media, and video transcript.

If none of those things are present on a post (like it's just an image), zero the embedding.

Also include a guard in the similarity search candidate generator that drops results if their embedding is empty. Empty embeddings are allowed for candidate generators that don't search on it, though.

**Comment by davidfreifeld** [comment]
hey @raindrift, when you wrote: "If none of those things are present on a post (like it's just an image), zero the embedding." I'm interpreting that as: don't pass an embedding to the ranker (or pass an all-zeroes embedding to the ranker). As opposed to zeroing the embedding at ingex time. Is that what you were thinking?
